[DPA-1361]: feat(deployment/mcms): update common changesets

graham-chainlink opened the pull request:
- Added new v2 version of set_config_mcms to support new MCMS library
- Added new v2 version of transfer_to_mcms_with_timelock to support new MCMS library

JIRA: https://smartcontract-it.atlassian.net/browse/DPA-1361

gustavogama-cll commented on deployment/common/changeset/set_config_mcms.go:
```diff
@@ -206,3 +210,75 @@ func SetConfigMCMS(e deployment.Environment, cfg MCMSConfig) (deployment.Changes
 }
+
+// SetConfigMCMSV2 is a reimplementation of SetConfigMCMS that uses the new MCMS library.
+func SetConfigMCMSV2(e deployment.Environment, cfg MCMSConfig) (deployment.ChangesetOutput, error) {
```
Just an early comment: @ecPablo (and also Rodrigo?) were planning on using a different folder layout for the changesets that rely on the new mcms lib.

See [this PR](https://github.com/smartcontractkit/chainlink/pull/16163) to understand what I mean (just the `mcmcnew` folder).

graham-chainlink replied:
Ah okay i didn know, happy to follow

gustavogama-cll commented on deployment/common/changeset/set_config_mcms.go:
```diff
@@ -206,3 +210,75 @@ func SetConfigMCMS(e deployment.Environment, cfg MCMSConfig) (deployment.Changes
+	inspectorPerChain := map[uint64]sdk.Inspector{}
+	proposerMcmsPerChain := map[uint64]string{}
+
+	mcmsStatePerChain, err := MaybeLoadMCMSWithTimelockState(e, selectors)
```
We need to review whether we need to update this when #16163 is merged.

graham-chainlink replied:
probably, if we use the new DeployMCMS changeset then we need to update this too to the one you have in your PR

gustavogama-cll commented on deployment/common/changeset/transfer_to_mcms_with_timelock.go:
```diff
@@ -144,6 +150,69 @@ func TransferToMCMSWithTimelock(
+		for _, contract := range contracts {
+			// Just using the ownership interface.
+			// Already validated is ownable.
+			owner, c, _ := LoadOwnableContract(contract, e.Chains[chainSelector].Client)
```
If I understood Pablo's comments earlier this week, this won't work for Solana. We'll need to modify this changeset significantly. And soon.

graham-chainlink replied:
yeh and probably to most changeset too as they are quite evm specific.

gustavogama-cll replied:
yeah, just realized it when I saw your previous PR.

graham-chainlink committed:
feat(deployment/mcms): update common changesets

- Added new v2 version of set_config_mcms to support new MCMS library
- Added new v2 version of transfer_to_mcms_with_timelock to support new MCMS library

JIRA: https://smartcontract-it.atlassian.net/browse/DPA-1361
